step-controller/step-controller-server/src/main/java/step/core/execution/ExecutionServices.java

```diff
@@ -72,6 +72,7 @@ public void init() {
 	@Path("/start")
 	@Secured(right="plan-execute")
 	public String execute(ExecutionParameters executionParams) {
+		checkRightsOnBehalfOf("plan-execute", executionParams.getUserID());

```

Comment:
@iegorov777 @david-stephan The UserID is set by the REST client. If UserID is different from the authenticated user, shouldn't we apply additional checks? we could for instance add a right that allows users to run executions on behalf?

Reply:
Just saw that this is implemented on EE side. You can ignore the previous comment